### PR TITLE
[Keyboard] Add Mixi

### PR DESCRIPTION
**Pull request by ohchiko**
## Description

Adds Mixi Macropad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by ohchiko** [comment]
Any other things to do to get this PR accepted?

File: keyboards/mixi/config.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4752
+#define PRODUCT_ID      0x4d49
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    gregorio <chiko150@outlook.com>
```
Comment (fauxpark) [inline keyboards/mixi/config.h:9]:
```suggestion
#define MANUFACTURER    gregorio
```

File: keyboards/mixi/rules.mk [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = no	    # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = no	    # Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes	    # Audio control and System control
+CONSOLE_ENABLE = yes	    # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = no	    	# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (fauxpark) [inline keyboards/mixi/rules.mk:23]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
NKRO_ENABLE = no            # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```

File: keyboards/mixi/keymaps/via/rules.mk [hunk 3]
```diff
@@ -0,0 +1,8 @@
+VIA_ENABLE = yes
+LINK_TIME_OPTIMIZATION_ENABLE = yes
+LEADER_ENABLE = no
+FAUXCLICKY_ENABLE = no
+MIDI_ENABLE = no
+BLUETOOTH_ENABLE = no
+KEY_LOCK_ENABLE = no
+TERMINAL_ENABLE = no
```
Comment (fauxpark) [inline keyboards/mixi/keymaps/via/rules.mk:8]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/mixi/rules.mk [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+ENCODER_ENABLE = yes
+AUDIO_ENABLE = no
+TAP_DANCE_ENABLE = no
```
Comment (fauxpark) [inline keyboards/mixi/rules.mk:28]:
```suggestion
```

File: keyboards/mixi/config.h (hunk 1 above)
Comment (ohchiko) [reply keyboards/mixi/config.h]: resolved on new commit

File: keyboards/mixi/rules.mk (hunk 2 above)
Comment (ohchiko) [reply keyboards/mixi/rules.mk]: resolved on new commit

File: keyboards/mixi/rules.mk (hunk 4 above)
Comment (ohchiko) [reply keyboards/mixi/rules.mk]: resolved on new commit

File: keyboards/mixi/keymaps/via/rules.mk (hunk 3 above)
Comment (ohchiko) [reply keyboards/mixi/keymaps/via/rules.mk]: resolved on new commit

**Comment by ohchiko** [comment]
oh sorry, i didn't know that re-requesting a review would remove another reviewer. my fault.

**Comment by ohchiko** [comment]
Would you like to add the `via` label too? Because this PR contains a VIA keymap.

File: keyboards/mixi/rules.mk [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = no        # Mouse keys(+4700)
```
Comment (fauxpark) [inline keyboards/mixi/rules.mk:18]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
```

File: keyboards/mixi/config.h [hunk 6]
```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4752
+#define PRODUCT_ID      0x4d49
```
Comment (fauxpark) [inline keyboards/mixi/config.h:7]:
```suggestion
#define PRODUCT_ID      0x4D49
```

File: keyboards/mixi/rules.mk (hunk 5 above)
Comment (ohchiko) [reply keyboards/mixi/rules.mk]: resolved on new commit

File: keyboards/mixi/config.h (hunk 6 above)
Comment (ohchiko) [reply keyboards/mixi/config.h]: resolved on new commit

File: keyboards/mixi/keymaps/via/rules.mk [hunk 7]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
```
Comment (KrezzWaskito) [inline keyboards/mixi/keymaps/via/rules.mk:1]: VIAL_ENCODERS_ENABLE = yes

**Pull request merged**